src/types.ts

```diff
@@ -170,3 +170,19 @@ type _TupleOf<
 > = R['length'] extends TSize
   ? R
   : _TupleOf<TNumber, TSize, readonly [TNumber, ...R]>
+
+export type IsEmptyObject<
+  T extends object,
+  AllKeys extends keyof T = keyof T,
+> = [AllKeys] extends [never] ? true : false
+
+export type Flatten<
```

Comment:
This also filters out all `never` types. It wasn't playing to well with `never` as members since `never extends any[]` is true.